Procurement/ViewModel/Recipes/RareSetRecipe.cs

```diff
@@ -19,23 +27,39 @@ public override string Name
 
         private List<MatchedSet> sets = new List<MatchedSet>();
         
-        public RareSetRecipe(int minimumItemLevel, int maximumItemLevel, bool itemsIdentified, string name)
+        public RareSetRecipe(int minimumItemLevel, int maximumItemLevel, bool itemsIdentified, string name, SetType setType = SetType.Normal)
             : base(80)
         {
             this.minimumItemLevel = minimumItemLevel;
             this.maximumItemLevel = maximumItemLevel;
             this.itemsIdentified = itemsIdentified;
             this.name = name;
+            _setType = setType;
         }
 
         public override IEnumerable<RecipeResult> Matches(IEnumerable<Item> items)
         {
             List<Gear> allGear = items.OfType<Gear>().ToList();
+            bool isShaperWanted = false;
```

Comment:
You could do something like the following and not need the switch statement.

``` C#
bool isShaperWanted = _setType == SetType.Shaper;
bool isElderWanted = _setType == SetType.Elder;
```

Alternatively, since these variables are only being used in one place, you could instead check them in the `Where` function, below.  But that might make the already-complicated `Where` function overly complex.

Reply:
I knew there was a better way! I had considered adding to the LINQ but it's already complex enough.